File: api/DTOs/HR/ChecklistHRDto.cs
using api.Entities.HR;

namespace api.DTOs.HR
{
    public class ChecklistHRDto
    {
        public int Id {get; set;}
        public int CandidateId { get; set; }
        public int OrderItemId { get; set; }
        public string UserName {get; set;}
        public string HrExecUsername { get; set; }
        public DateTime CheckedOn {get; set;}
        public int Charges {get; set;}
        public int ChargesAgreed { get; set; }
        public bool ExceptionApproved {get; set;}
        public string ExceptionApprovedBy {get; set;}
        public DateTime ExceptionApprovedOn {get; set;}
        public string SalaryOffered {get; set;}
        public bool ChecklistedOk {get; set;}
        public bool RequireInternalReview {get; set;}
        //following members additional
        public int ApplicationNo { get; set; }  //new
        public string CandidateName { get; set; } //new
        public string CategoryRef { get; set; } 
        public string OrderRef { get; set; }
        public bool AssessmentIsNull {get; set;}
        
        public ICollection<ChecklistHRItem> ChecklistHRItems {get; set;}
    }
}

File: client/src/app/_dtos/hr/checklistHRDto.ts
import { IChecklistHRItem } from "src/app/_models/hr/checklistHRItem";


export interface IChecklistHRDto{

     id: number;
     candidateId: number;
     applicationNo: number;
     candidateName: string;
     categoryRef: string;
     orderRef: string;
     orderItemId: number;
     userLoggedId: number;
     userLoggedName: string;
     hrExecUsername: string;
     checkedOn: Date;
     hrExecComments: string;
     charges: number;
     chargesAgreed: number;
     exceptionApproved: boolean;
     exceptionApprovedBy: string;
     exceptionApprovedOn: Date;
     checklistedOk: boolean;
     assessmentIsNull: boolean;
     requireInternalReview: boolean;
     checklistHRItems: IChecklistHRItem[];
}

export class ChecklistHRDto implements IChecklistHRDto{

     id: number=0;
     candidateId: number=0;
     applicationNo: number=0;
     orderItemId: number=0;
     categoryRef: string='';
     candidateName: string='';
     orderRef: string='';
     userLoggedId: number=0;
     userLoggedName: string='';
     hrExecUsername: string = '';
     checkedOn: Date = new Date();
     hrExecComments: string='';
     charges: number=0;
     chargesAgreed: number=0;
     exceptionApproved: boolean=false;
     exceptionApprovedBy: string='';
     exceptionApprovedOn: Date=new Date();
     checklistedOk: boolean=false;
     assessmentIsNull = false;
     requireInternalReview=false;
     checklistHRItems: IChecklistHRItem[]=[];
}